### PR TITLE
RecoLocalTracker/SiPixelRecHits: change return type of ESProducer's to unique_ptr.

### PR DESCRIPTION
**Pull request by gartung**

### CODE REVIEW

**Comment by gartung** [comment]
@Dr15Jones

**Comment by Dr15Jones** [comment]
please test

File: RecoLocalTracker/SiPixelRecHits/plugins/PixelCPETemplateRecoESProducer.cc [hunk 1]
```diff
@@ -60,9 +60,8 @@ PixelCPETemplateRecoESProducer::produce(const TkPixelCPERecord & iRecord){
   ESHandle<SiPixelTemplateDBObject> templateDBobject;
   iRecord.getRecord<SiPixelTemplateDBObjectESProducerRcd>().get(templateDBobject);
 
-  //  cpe_  = std::make_shared<PixelCPETemplateReco>(pset_,magfield.product(),lorentzAngle.product(),templateDBobject.product() );
-  cpe_  = std::make_shared<PixelCPETemplateReco>(pset_,magfield.product(),*pDD.product(),*hTT.product(),lorentzAngleProduct,templateDBobject.product() );
-  return cpe_;
+  //  cpe_  = std::make_unique<PixelCPETemplateReco>(pset_,magfield.product(),lorentzAngle.product(),templateDBobject.product() );
```
Comment (slava77) [inline RecoLocalTracker/SiPixelRecHits/plugins/PixelCPETemplateRecoESProducer.cc]: please remove the commented code or update it to match the latest update ```//   return std::make_unique ...```

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/22031	7acd495
- technical
- jenkins tests pass

**Comment by fabiocos** [comment]
+1

**Comment by pmaksim1** [comment]
For the record: this was lost, then found, then reimplemented in #22458.  Sorry about that.

**Pull request merged**